app/controllers/pay_controller.js

```diff
@@ -52,19 +54,23 @@ module.exports.bindRoutesTo = (app) => {
       }
     };
 
-  console.log('Making a payment to: ' + payApiUrl);
-    client.post(payApiUrl, paymentRequest, (data, payApiResponse) => {
+    logger.info('Making a payment to: ' + PAY_API_URL);
+    client.post(PAY_API_URL, paymentRequest, (data, payApiResponse) => {
       logger.info('pay api response: ', data);
       if (payApiResponse.statusCode == 201) {
+
         req.state[paymentReference] = { pid: data.payment_id };
```

Comment:
Yes, we can setup as many payments as we want which will result in one pid for each. lastPayment is needed just because we need to identify which one of those pid is the last one.